analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -580,6 +592,11 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
         )
     }
 
+    /**
+     * Convenience function to obtain the [MavenRepositorySystem] from the container.
+     */
+    private fun fetchMavenRepositorySystem() = container.lookup(MavenRepositorySystem::class.java, "default")
```

Comment:
How about generalizing this like

    private inline fun <reified T> lookup(hint: String = "default") = container.lookup(T::class.java, hint)

and making use of it in *all* `container.lookup()` calls like

    val populator = lookup<MavenExecutionRequestPopulator>()

Ideally, that code simplification would go into a separate first commit as it's not strictly related to the actual fix.

Reply:
Great idea. I implemented the generic lookup function. I made it public, however, as the Maven class also makes a lookup.